cypress/e2e/invoices/pay-invoice-for-previous-fy.cy.js

```diff
@@ -0,0 +1,255 @@
+        });
+      });
+
+    cy.createTempUser([
+      Permissions.uiFinanceViewFundAndBudget.gui,
+      Permissions.uiInvoicesApproveInvoices.gui,
+      Permissions.viewEditCreateInvoiceInvoiceLine.gui,
+      Permissions.uiInvoicesPayInvoices.gui,
+      Permissions.uiInvoicesPayInvoicesInDifferentFiscalYear.gui,
+      Permissions.uiOrdersView.gui,
+    ]).then((userProperties) => {
+      testData.user = userProperties;
+
+      cy.login(userProperties.username, userProperties.password, {
+        path: TopMenu.ordersPath,
+        waiter: Orders.waitLoading,
+      });
+    });
+  });
+
+  after('Delete test data', () => {
+    Organizations.deleteOrganizationViaApi(testData.organization.id);
+    ServicePoints.deleteViaApi(testData.servicePoint.id);
+    // Orders.deleteOrderViaApi(testData.order.id);
+    // FiscalYears.deleteFiscalYearViaApi(fiscalYears.prev.id);
+    FiscalYears.deleteFiscalYearViaApi(fiscalYears.next.id);
+    // Budgets.deleteBudgetWithFundLedgerAndFYViaApi(testData.budget);
+    // InventoryHoldings.deleteHoldingRecordByLocationIdViaApi(testData.location.id);
+    // Locations.deleteViaApi(testData.location);
+    Users.deleteViaApi(testData.user.userId);
+  });
+
+  it(
+    'C388520 Approve and pay invoice created in current FY for previous FY when related order line was created in previous FY (thunderjet) (TaaS)',
+    { tags: [TestTypes.criticalPath, DevTeams.thunderjet] },
+    () => {
+      // Click on "PO number" link on "Orders" pane
+      const OrderDetails = Orders.selectOrderByPONumber(testData.order.poNumber);
+      OrderDetails.checkOrderStatus(ORDER_STATUSES.OPEN);
+
+      // Click "Actions" button, Select "New invoice" option, Click "Submit" button
+      OrderDetails.createNewInvoice();
+
+      // * "Create vendor invoice" page appears
+      // * "Vendor name" field is not editable and contains vendor name from related order from Preconditions item #4
+      InvoiceEditForm.checkButtonsConditions([
+        {
+          label: 'Vendor name',
+          conditions: { disabled: true, value: testData.invoice.vendorName },
+        },
+      ]);
+
+      // Fill the following fields:
+      // * "Invoice date" - select any date within "Fiscal year #2" period (e.g. today's)
+      // * "Fiscal year" - select previous "Fiscal year #1"
+      InvoiceEditForm.fillInvoiceFields({ ...testData.invoice, vendorName: undefined });
+
+      // Click "Save & close" button
+      InvoiceEditForm.clickSaveButton({ invoiceCreated: true, invoiceLineCreated: true });
+
+      // Fiscal year" field is specified with previous "Fiscal year #1"
+      Invoices.checkCreatedInvoice({
+        ...testData.invoice,
+        invoiceNumber: testData.invoice.vendorInvoiceNo,
+        information: { fiscalYear: fiscalYears.prev.code, status: 'Open' },
+      });
+      // #6 Click "Actions" button on the third "Vendor invoice number - <number entered in Step #4>" pane -> select "Approve & pay" option
```

Comment:
Does we need so much comment code for AI?

Reply:
This tests is not fully implemented yet. Once work will be completed it will contain much less commented code.